src/tests/ftest/get_remote_files.sh

```diff
@@ -0,0 +1,369 @@
+# Arguments:
+#   $1: local logs
+# Returns:
+#   0: All cart_logtest were successful
+#   1: Failure detected with cart_logtest
+#######################################
+run_cartlogtest() {
+    rc=0
+    if [ -f "${CART_LOGTEST_PATH}" ]; then
+        # shellcheck disable=SC2045,SC2086
+        for file in $(ls -d ${1})
+        do
+            if [ -f ${file} ]; then
+                logtest_log="${file%.*}.cart_logtest.${file##*.}"
+                ${CART_LOGTEST_PATH} ${file} > ${logtest_log} 2>&1
+                ret=$?
+                if [ ${ret} -ne 0 ]; then
+                    echo "  Error: details in ${file}_cart_testlog"
+                    rc=1
+                fi
+            fi
+        done
+    else
+        echo "  Error: ${CART_LOGTEST_PATH} does not exist!"
+        rc=1
+    fi
+    return ${rc}
+}
+
+####### MAIN #######
+if [[ $# -eq 0 ]] ; then
+    usage
+    exit 1
+fi
+
+# Setup defaults
+ARCHIVE_DEST=""
+FILES_TO_PROCESS=""
+COMPRESS="false"
+VERBOSE="false"
+CART_LOGTEST="false"
+EXCLUDE_ZIP="false"
+THRESHOLD=""
+rc=0
+
+# Step through arguments
+while getopts "vhxzca:f:t:" opt; do
+    case ${opt} in
+        a )
+            ARCHIVE_DEST=$OPTARG
+            ;;
+        f )
+            FILES_TO_PROCESS=$OPTARG
+            ;;
+        t )
+            THRESHOLD=$OPTARG
+            ;;
+        c )
+            CART_LOGTEST="true"
+            ;;
+        z )
+            COMPRESS="true"
+            ;;
+        x )
+            EXCLUDE_ZIP="true"
+            ;;
+        v )
+            VERBOSE="true"
+            ;;
+        h )
+            usage
+            exit
+            ;;
+        ? )
+            echo "Unexpected argument: ${opt}!"
+            usage
+            exit
+            ;;
+    esac
+done
+shift "$((OPTIND -1))"
+
+script_name=$(basename "${0}")
+log_file="/tmp/${script_name%%.*}_$(/bin/date +%Y%m%d_%H%M%S).log"
+if [ "${VERBOSE}" != "true" ]; then
+    # Redirect 'set -e' output to a log file if not verbose
+    exec 2> "${log_file}"
+fi
+set -ex
+echo "Logging ${script_name} output to ${log_file}"
+
+# Verify files have been specified and they exist on this host
+check_files_input "${FILES_TO_PROCESS}"
+
+# Display big files to stdout
+if [ -n "${THRESHOLD}" ]; then
+    echo "Checking if files exceed ${THRESHOLD} threshold ..."
+    list_tag_files "${FILES_TO_PROCESS}" "${THRESHOLD}"
+fi
+
+# Run cart_logtest.py on FILES_TO_PROCESS
+if [ "${CART_LOGTEST}" == "true" ]; then
+    echo "Running ${CART_LOGTEST_PATH} ..."
+    run_cartlogtest "${FILES_TO_PROCESS}"
+    ret=$?
+    if [ ${ret} -ne 0 ]; then
+        rc=1
+    fi
+fi
+
+# Compress files in FILES_TO_PROCESS if not running on VM host.
+if [ "${COMPRESS}" == "true" ]; then
+    echo "Compressing files larger than 1M ..."
+    if check_hw "${EXCLUDE_ZIP}"; then
+        echo "VM system detected, skipping compression ..."
+    else
+        compress_files "${FILES_TO_PROCESS}"
+        ret=$?
+        if [ ${ret} -ne 0 ]; then
```

Comment:
This won't work with `set -e` enabled as the script will exit if `compress_files` exits `!0`:

```suggestion
        if ! compress_files "${FILES_TO_PROCESS}"; then
```